src/rgw/rgw_op.cc

```diff
@@ -3273,13 +3275,20 @@ static CompressorRef get_compressor_plugin(const req_state *s,
   const auto& upload_id = s->info.args.get("uploadId", &is_multipart);
 
   if (!is_multipart) {
-    return Compressor::create(s->cct, compression_type);
-  }
+    int count = 0;
+recreate:
+    compressor = Compressor::create(s->cct, compression_type);
+    if ((compressor == nullptr) && (++count <= 3))
```

Comment:
Frankly speaking, it's a bit arbitrary to set it to 3. 

The main reason why magic number is introduced is not to get a good result but to avoid deadlock by other reason except the random choose, e.g zlib *.so is not in the proper path.

Reply:
@fangyuxiangGL Why do we even retry attempts to create compressor? Does it have some transient creation problems?